src/views/blocks/rsvp/form.php

```diff
@@ -1,37 +1,51 @@
+ * @version TBD
  *
+ * @var Tribe__Tickets__Editor__Template $this                Template object.
+ * @var int                              $post_id             [Global] The current Post ID to which RSVPs are attached.
+ * @var array                            $attributes          [Global] RSVP attributes (could be empty).
+ * @var Tribe__Tickets__Ticket_Object[]  $active_rsvps        [Global] List of RSVPs.
+ * @var bool                             $all_past            [Global] True if RSVPs availability dates are all in the past.
+ * @var bool                             $has_rsvps           [Global] True if the event has any RSVPs.
+ * @var bool                             $has_active_rsvps    [Global] True if the event has any RSVPs available.
+ * @var bool                             $must_login          [Global] True if only logged-in users may obtain RSVPs.
+ * @var string                           $login_url           [Global] The site's login URL.
+ * @var int                              $threshold           [Global] The count at which "number of tickets left" message appears.
+ * @var null|string                      $step                [Global] The point we're at in the loading process.
+ * @var bool                             $opt_in_checked      [Global] Whether appearing in Attendee List was checked.
+ * @var string                           $opt_in_attendee_ids [Global] The list of attendee IDs to send in the form submission.
+ * @var string                           $opt_in_nonce        [Global] The nonce for opt-in AJAX requests.
+ * @var bool                             $doing_shortcode     [Global] True if detected within context of shortcode output.
+ * @var bool                             $block_html_id       [Global] The RSVP block HTML ID. $doing_shortcode may alter it.
+ * @var Tribe__Tickets__Ticket_Object    $ticket              The ticket object with provider set to RSVP.
+ * @var string                           $going               The RSVP status at time of add/edit, or empty if not in that context.
  */
-
-$going      = $this->get( 'going' );
-$must_login = ! is_user_logged_in() && tribe( 'tickets.rsvp' )->login_required();
 ?>
-<!-- This div is where the AJAX returns the form -->
-<div class="tribe-block__rsvp__form">
-	<?php if ( ! empty( $going ) && ! $must_login ) :
-		$ticket = $this->get( 'ticket' );
-		$args = [
-			'ticket_id' => $ticket->ID,
-			'post_id'   => $post_id,
-			'ticket'    => tribe( 'tickets.rsvp' )->get_ticket( get_the_id(), $ticket->ID ),
-			'going'     => esc_html( $going ),
-		];
 
-		// can't escape, contains html
-		echo tribe( 'tickets.editor.template' )->template( 'blocks/rsvp/form/form', $args, false );
-	endif; ?>
+<!-- This div is where the AJAX returns the form. -->
+<div class="tribe-block__rsvp__form">
+	<?php
+	if (
+		! empty( $going )
+		&& ! $must_login
+	) :
+		// Can't escape, contains HTML.
+		echo $this->template( 'blocks/rsvp/form/form', [], false );
```

Comment:
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$this'.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '['
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'false'

<sub>hash: 83a34cf63bc9c5f132245e5791dd63f4</sub>